pkg/cloudprovider/aws/instanceprofile.go

```diff
@@ -56,14 +57,17 @@ func (p *InstanceProfileProvider) Get(ctx context.Context, cluster *v1alpha1.Clu
 }
 
 func (p *InstanceProfileProvider) getInstanceProfile(ctx context.Context, cluster *v1alpha1.ClusterSpec) (*iam.InstanceProfile, error) {
-	output, err := p.iam.GetInstanceProfileWithContext(ctx, &iam.GetInstanceProfileInput{
+	timeoutCtx, cancel := context.WithTimeout(ctx, time.Duration(300) * time.Second)
```

Comment:
This feels like a fair amount of duplication - eg it'll be hard to change later. I'm wondering if this is the right level to do this kind of timeout-setting, or if this function should just assume a `Context` gets passed in that has the right (overall) timeout set?